Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -396,6 +396,8 @@ def uninstall_rmdir(*directories, command: nil, **_)
               command.run!("/bin/rm", args: ["-f", "--", ds_store], sudo: true, print_stderr: false)
             end
 
+            uninstall_rmdir(*Dir["#{resolved_path}/*/"], command: command) unless Dir.empty?(resolved_path)
+
             command.run("/bin/rmdir", args: ["--", resolved_path], sudo: true, print_stderr: false)
```

Comment:
Maybe add `.success?` so we can return early if the recursive call returns `false`.